### ISSUE TITLE
html problems

### ISSUE THREAD

**Issue by boredland** [DIYBookScanner/spreads#172]
Hi there,

If got severe display-problems. I testes in Full-HD and HD-Ready resolutions, the phenomenon is the same:
1. The preview Lightbox is bigger than the displayed page.
https://qsnapnet.com/snaps/9fg5trw3wxmt2o6
2. Klicking in the border-area will result in opening the preview-image in a new tab. Only escape brings you back. Annoying if using the browser in fullscreen.
https://qsnapnet.com/snaps/138aaqwmr2ro1or
3. The Cropping-Window is bigger than the displayed page. No chance to klick enter.
https://qsnapnet.com/snaps/2lnvuu7pty44pld

**Comment by jbaiter** [comment]
Yep, it's a mess.... Do you know JavaScript/CSS and would like to help?
I don't have a lot of time at the moment, and I'd be *really* glad about some help.

**Comment by adongy** [comment]
I could do some JS/CSS but not having a camara to test it is pretty hard :( Can we use a fake workflow and still use the web ui?

**Comment by boredland** [comment]
Jep, just tested that yesterday. We could start a branch for css-correction
together to not annoy johannes and tell him when it's finished.

2015-03-26 15:53 GMT+01:00 Anthony Dong <notifications@github.com>:

**Comment by jbaiter** [comment]
adongy, you totally can, that's actually the way I do most of my development. Just use the "dummy" device driver and you should be good to go!

**Comment by knjigor** [comment]
Hi, I'm a web designer, mostly I do css (less js) but if someone could point me where to look maybe I could do something.

**Comment by adongy** [comment]
Hey @knjigor ! The client web interface is located at https://github.com/DIYBookScanner/spreads/tree/master/spreadsplug/web/client , so you can always start looking at the scss folder.

**Comment by knjigor** [comment]
I have cloned spreads from git `https://github.com/DIYBookScanner/spreads.git` (spreads 0.5git20150515.3712) and noticed that when I install it and try to run `spread web` I get only white page, no errors, nothing. It is working when I install latest version from buildbot `http://buildbot.diybookscanner.org/nightly/spreads-latest.tar.gz`,   spreads-0.5git20150514.3712

**Comment by jbaiter** [comment]
@knjigor, you need to build the JavaScript bundles first. Go to `spreadsplug/web/client` and run `make watch`, this will install all the dependencies, build the development bundle and rebuild whenever you change any of the JavaScript or SCSS sources.

However, I'm currently working on a huge refactoring of the web interface that removed a lot of unneccessary complexity by relying on some of the ecosystem that has evolved around React over the last year, namely [Bootstrap](http://react-bootstrap.github.io/index.html) instead of Foundation for the UI components and [Alt](https://goatslacker.github.io/alt/) instead of Backbone for the data flow. It's currently being developed on this branch of my personal fork: https://github.com/jbaiter/spreads/tree/alt-refactor

I'm pretty far into the refactoring and should be able to merge it back into the master branch in a few weeks at the latest. You're welcome to dig through the code right now, though and ask me questions :-)
Styling should become a whole lot easier, too, since you're probably more familiar with Bootstrap anyways and there's a whole lot more custom components for it out there than for Foundation.

**Comment by knjigor** [comment]
@jbaiter Ok, thanx for tip, I will clone your fork and look at it. Unfortunately, i have some problems, when i try to run `make watch` in  `spreadsplug/web/client` it finish with error, here is terminal log http://pastebin.com/VdwTvsn4

**Comment by knjigor** [comment]
I have changed version of node.js with this installation 
``curl -sL https://deb.nodesource.com/setup | sudo bash -
sudo apt-get install nodejs``

and get to:
``
 [196] ./~/react/lib/getEventKey.js 3505 {0} [built]
 [197] ./~/react/lib/isNode.js 1022 {0} [built]
 [198] ./~/react/lib/getNodeForCharacterOffset.js 1956 {0} [built]
 [199] ./~/react/lib/Danger.js 7283 {0} [built]
 [200] ./~/react/lib/createNodesFromMarkup.js 2967 {0} [built]
 [201] ./~/react/lib/getMarkupWrap.js 3414 {0} [built]
 [202] ./~/react/lib/createArrayFrom.js 2637 {0} [built]
 [203] ./~/react/lib/toArray.js 2365 {0} [built]
``
and it stop there, no error just stay.

**Comment by knjigor** [comment]
@jbaiter When I clone your fork it also get stuck http://pastebin.com/xjN8JBEd

**Comment by knjigor** [comment]
Hello guys, I managed to find an interim solution for crop page problem, I have changed two lines in 
``spreadsplug/web/client/scss/app.scss``
```
 img {
    width: 25%;
    //max-height: 70%;
```
It doesn't look extra nice but it is functional, I have tested it in   192x1080 and 1680x945 resolution.
Problem is that it isn't responsive.

**Comment by knjigor** [comment]
Preview lightbox fix
``spreadsplug/web/client/scss/app.scss``
```
.overlay, .overlay .lightbox a {
    align-items: center;
    display: inline-flex;
    justify-content: center;
}
.overlay .lightbox img {
    image-orientation: from-image;
    width: 32%;
}
```
to close lightbox without opening image in new window click just above or below image

**Comment by knjigor** [comment]
Has anyone tried this? Just to know if I'm going in right directions?

**Comment by boredland** [comment]
Closing this myself as the reporter — I'm no longer pursuing it, and I'd rather not leave it sitting in your triage queue.

No action is needed from the maintainers, and this says nothing about the validity of the report: if anyone else is hitting the same thing, please reopen or file fresh.

**Comment by boredland** [comment]
Reopening — please disregard my previous comment.

That closure was a mistake on my side: it came from a bulk sweep of my own filed issues, not from a considered decision to withdraw this report. Sorry for the noise in your notifications.

The report stands as originally written.
